### PR TITLE
Disable UI rendering when history is skipped

### LINKED ISSUE #558: Decouple UI and canvas
Right now, whenever we drag, we call setState() which calls render() on the App component which re-renders the UI and on componentDidUpdate updates the canvas.

This is wasteful because the UI doesn't change and we're triggering the entire re-render. Especially now that we are displaying a sizable UI.

We should move all the UI in its own component and shield it under memoized / PureComponent / shouldComponentUpdate.

### PR DESCRIPTION
**Pull request by vjeux**
When we are scrolling, resizing, or moving elements, we already disable the history. Since those actions do not change the state of the UI, we can also avoid re-drawing it and save ~10ms per frame.

I had to change all the forceUpdate() to setState({}), otherwise it would bypass shouldComponentUpdate.

Fixes #558

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -286,7 +294,7 @@ export class App extends React.Component<any, AppState> {
   private onKeyDown = (event: KeyboardEvent) => {
     if (event.key === KEYS.ESCAPE) {
       elements = clearSelection(elements);
-      this.forceUpdate();
+      this.setState({});
```
Comment (Fausto95) [inline src/index.tsx]: This one might not be needed since we're already  calling setState on the next line

**Comment by dwelle** [comment]
Are we sure that `this.setState({})` makes the intent clearer than `this.forceUpdate()`? Or is the idea to batch updates? Don't recall when updates are batched in classical react (event handlers only?).

**Comment by vjeux** [comment]
I'm going to merge it in since it's such a big performance boost. Happy to do any follow up.

**Comment by vjeux** [comment]
@dwelle here's a small codesandbox that shows the difference in behavior between setState and forceUpdate: https://codesandbox.io/s/vigorous-kowalevski-9p3k2

**Comment by dwelle** [comment]
Cool. Yea, I don't think there's any problem with the PR, I just didn't have time to do a proper review the way I do it :).

**Pull request merged**